app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java

```diff
@@ -236,7 +254,7 @@ public void onFragmentFirstTimeVisible() {
                 final int lno = args.getInt(Document.EXTRA_FILE_LINE_NUMBER);
                 if (lno >= 0) {
                     startPos = StringUtils.getIndexFromLineOffset(_hlEditor.getText(), lno, 0);
-                } else {
+                } else if (lno == Document.EXTRA_FILE_LINE_NUMBER_LAST){
```

Comment:
Using a sentinel with a random value (not -1) to explicitly trigger the open file at end case.

Fixes the bug with shortcuts opening files at the end. Will break this out into a separate PR